lib/src/zap/zap.h

```diff
@@ -554,6 +559,37 @@ zap_err_t zap_close(zap_ep_t ep);
  */
 zap_err_t zap_send(zap_ep_t ep, void *buf, size_t sz);
 
+/**
+ * \brief Send data to peer using map.
+ *
+ * This is a more efficient send interface comparing to \c zap_send(). The data
+ * to be sent is described by \c map, \c buf and \c len. The data will not be
+ * copied out to the internal zap buffer (unlike \c zap_send()) and the
+ * application should not modify the data in the specified range (buf[0..len-1])
+ * before the send operation is completed. The completion of the send operation
+ * results in \c ZAP_EVENT_SEND_COMPLETE event with the specified \c context
+ * delivered via the callback function specified in \c zap_new() or \c
+ * zap_accept(). The completion status could be success or failed.
+ *
+ * If the function call returns \c ZAP_ERR_OK, it is guaranteed that \c
+ * ZAP_EVENT_SEND_COMPLETE corresponding to the call will be delivered. On the
+ * other hand, if this function returns an error (synchronously failed), it is
+ * also guaranteed that \c ZAP_EVENT_SEND_COMPLETE (that would associate with
+ * the call) won't be delivered.
+ *
+ * \param ep      The endpoint handler.
+ * \param map     The map handler describing the memory region used in the send
+ *                operation.
+ * \param buf     The pointer of the send buffer.
+ * \param len     The length of the data to be sent.
+ * \param context The application context coupling with the completion event.
```

Comment:
Remove "coupling with the completion event"

Reply:
done